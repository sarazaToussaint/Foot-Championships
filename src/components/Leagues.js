import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import { Link } from 'react-router-dom';
import { getLeagues } from './redux/leages';
import classes from './Leagues.module.css';

const Leagues = () => {
  const leagues = useSelector((state) => state.leagues);
  // const details = useSelector((state) => state.details);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getLeagues());
  }, [dispatch]);

  return (
    <>
      <div className="grid-container">
        <div>
          <h2>WELCOME TO TOP 20 FIFA LEAGUES</h2>
        </div>
        <div className={classes.leagues}>
          {leagues.map((league, index) => (
            <div className={(index % 2 === 0) ? 'rowbg' : ''} key={league.id}>
              <Link key={league.id} to={`/${league.id}`} state={{ id: league.id }}>
                {index}
                <div className={classes.league} key={league.id}>
                  <div>
                    <img src={league.logo} alt={league.name} className={classes.image} />
                  </div>
                  <div className={classes.upperLeague}>
                    <BiRightArrowCircle
                      className={classes.leagueBtn}
                      style={{
                        width: '25px',
                        height: '25px',
                        color: '#fff',
                        background: 'transparent',
                      }}
                    />
                    <p className={classes.leagueName}>{league.name}</p>
                  </div>
                </div>
              </Link>
            </div>
          ))}
        </div>
      </div>
    </>
  );
};

export default Leagues;
