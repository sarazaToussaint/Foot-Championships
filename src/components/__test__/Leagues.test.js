import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configStore';
import Leagues from '../Leagues';

describe('Leagues test', () => {
  it('renders leagues', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Leagues />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
