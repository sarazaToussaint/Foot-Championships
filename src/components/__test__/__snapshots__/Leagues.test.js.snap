// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues test renders leagues 1`] = `
<div
  className="grid-container"
>
  <div>
    <h2>
      WELCOME TO TOP 20 FIFA LEAGUES
    </h2>
  </div>
  <div
    className="leagues"
  />
</div>
`;
